Gems/ROS2/Code/CMakeLists.txt

```diff
@@ -0,0 +1,158 @@
+# Copyright (c) Contributors to the Open 3D Engine Project.
+# For complete copyright and license terms please see the LICENSE at the root of this distribution.
+#
+# SPDX-License-Identifier: Apache-2.0 OR MIT
+
+# If no ROS2 is found, no targets are valid for this Gem
+find_package(ROS2 MODULE)
+if (NOT ROS2_FOUND)
+    return()
+endif()
+
+# Add the ROS2.Static target
+# Note: We include the common files and the platform specific files which are set in ros2_common_files.cmake
+# and in ${pal_dir}/ros2_${PAL_PLATFORM_NAME_LOWERCASE}_files.cmake
+ly_add_target(
+    NAME ROS2.Static STATIC
+    NAMESPACE Gem
+    FILES_CMAKE
+        ros2_files.cmake
+    INCLUDE_DIRECTORIES
+        PUBLIC
+            Include
```

Comment:
Currently, `ROS2Bus.h` is being included as part of the STATIC target ROS2.Static. So if a project(or any other Gem) wants to use the `ROS2RequestBus` or the `ROS2Interface`, they would need to add either `ROS2.Static` or `ROS2` module(since that also adds Include as a public include dir) as a build dependency which would result in linking those unnecessarily. If they link against `ROS2` module instead of static, it's probably not a big deal but it could be avoided.
There's also the issue of installed(SDK) version of the engine [propagating private build dependencies as runtime dependencies](https://github.com/o3de/o3de/pull/5581). So if some other gem `MyGem` has a private build dependency on `ROS2`, and the project has both `MyGem` and `ROS2` gem enabled, we would end up trying to load both `ROS2.Editor` and `ROS2` modules as runtime dependencies which would lead to a bug similar to [this](https://github.com/o3de/o3de/issues/5822). This is probably an unlikely scenario but it's something to consider.
To avoid this, we could move the `ROS2Bus.h` to a header only target called `ROS2.API` and any gem/project that needs access can just add that as a build dependency. We do something similar for [ScriptCanvas](https://github.com/o3de/o3de/pull/5923) and [LmbrCentral](https://github.com/o3de/o3de/pull/6012) gems.
Not sure if this is relevant to this gem since I don't know how it will end up being used. But I thought it might be useful to consider since we've run into problems like this before.

Reply:
I have added such API target in the most recent changeset